lib/cfg-lexer.c

```diff
@@ -226,6 +226,27 @@ cfg_lexer_lookup_keyword(CfgLexer *self, YYSTYPE *yylval, YYLTYPE *yylloc, const
   return LL_IDENTIFIER;
 }
 
+void
+cfg_lexer_clear_include_level(CfgLexer *self, CfgIncludeLevel *level)
+{
+  g_free(level->name);
+  if (level->yybuf)
+    _cfg_lexer__delete_buffer(level->yybuf, self->state);
+
+  if (level->include_type == CFGI_FILE)
+    {
+      if (level->file.include_file)
+        fclose(level->file.include_file);
+      g_slist_foreach(level->file.files, (GFunc) g_free, NULL);
+      g_slist_free(level->file.files);
+    }
+  else if (level->include_type == CFGI_BUFFER)
+    {
+      g_free(level->buffer.content);
+    }
+  memset(level, 0, sizeof(*level));
```

Comment:
As I see this patch is just a refactor, except this line, which is a new addition. Can you explain the purpose of this line? If needed, probably better do outside the function. For now this function modifies her input parameter.